### PR TITLE
ENG-1596: Adds StorybookJS to common

### PR DESCRIPTION
**Pull request by agiron123**
## Describe your changes and why you are making these changes
This PR adds StorybookJS to @aqueducthq/common.

StorybookJs is a development tool which enables you to develop frontend UI components in isolation. It also serves as a great tool for documenting frontend components and how they should be used.

For more information, you can check out their home page at: https://storybook.js.org/.

## Getting Started:
From src/ui/common, run the following command:
```
npm run storybook
```

After bundling completes, you should be able to open your browser to http://localhost:6006 and see a live component playground.

I recommend watching the demo below for a quick tour of storybook and the features it offers.

## Related issue number (if any)
ENG-1596

## Loom demo (if any)
https://www.loom.com/share/cd353f4fc21246a996ca10032f6d385c

## Caveats / Quirks
When setting up storybook on our project, I ran into a couple quirks. 
1. When installing dependencies for common, please use:
```
npm install --force
```

2. When running storybook, a message appears saying that there is an upgrade
available (version 6.5.12 -> 6.5.13). Don't update this, the new version has an
issue with running npm versions older than npm 7, and we're using npm 8.5.0.

3. Storybook uses webpack 5 to build and run, wheras our project on common uses
vite. There is a storybook builder and runner available that supports vite, but
it's currently an experimental package and doesn't play so nicely with our setup.

I've spent a couple hours trying to upgrade and set things up with vite, but I wasn't
able to have any luck.

This 6.5.12 setup works quite well and supports live reload, and should serve our needs
for quite a while as a development tool.

## Checklist before requesting a review
- [x] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [x] I have performed a self-review of my code.
- [x] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [x] If this is a new feature, I have added unit tests and integration tests.
- [x] I have run the integration tests locally and they are passing.
- [x] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [x] All features on the UI continue to work correctly.
- [x] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: src/ui/common/src/stories/Button.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import { ComponentMeta, ComponentStory } from '@storybook/react';
```
Comment (likawind) [inline src/ui/common/src/stories/Button.stories.tsx:1]: what's the relationship between these component with those in `primitives`? Which ones should I use?

**Review by likawind — APPROVED** [review]
The demo's really cool! Do we have a follow up task on documentation? I'm curious about how this integrate with our d2d work

File: src/ui/common/src/stories/Button.stories.tsx (hunk 1 above)
Comment (agiron123) [reply src/ui/common/src/stories/Button.stories.tsx:1]: First off, this component is just here to demonstrate how StorybookJS works. 
The button component here comes from the default installation of StorybookJS.

So, we shouldn't use these over the components that are defined in primitives. Once I get some stories working for our own components we can come back and delete this component along with the other default stories and components that Storybook gives us.

**Comment by agiron123** [comment]
> The demo's really cool! Do we have a follow up task on documentation? I'm curious about how this integrate with our d2d work

@likawind as we begin to add more components, we should start adding stories for them. In terms of followup work, a nice place to start is to get basic stories up and going for the custom components that we've created. I'd prioritize newer components since they are probably least tested.

As far as your day to day work is concerned, Storybook should be used as a place for very quick iteration of UI components that allows you to develop the UI components in isolation. I recommend starting here and building out reusable components before adding them to a page or layout. 

There's more we can do like integration testing, snapshot testing and even connecting to figma with the storybook connect plugin.

I recommend reading through their docs a bit more to learn about what this tool is capable of and watching some of the youtube videos here for a demo:
https://www.youtube.com/c/StorybookJS/featured

**Pull request merged**